src/modules/NetworkController/private/Copy-CertificatesToFabric.ps1

```diff
@@ -0,0 +1,142 @@
+function Copy-CertificatesToFabric {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [System.IO.DirectoryInfo]$CertPath,
+
+        [Parameter(Mandatory = $true)]
+        [System.Security.SecureString]$CertPassword,
+
+        [Parameter(Mandatory = $true)]
+        [System.Object]$FabricDetails,
+
+        [Parameter(Mandatory = $false)]
+        [Switch]$RotateNodeCertificates,
+
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty
+    )
+
+    begin {
+        $certificateCache = @()
+        $certificateConfig = @{
+            NetworkController = @{}
+        }
+
+        $nodesToRotateRestCert = @()
+        $nodesToRotateRestCert += $FabricDetails.NetworkController
+        $nodesToRotateRestCert += $FabricDetails.SoftwareLoadBalancer
+        $nodesToRotateRestCert += $FabricDetails.Server
+
+        [System.String]$certDir = "$(Get-WorkingDirectory)\RotateCert"
+
+        "Scanning certificates within {0}" -f $CertPath.FullName | Trace-Output
+        $pfxFiles = Get-ChildItem -Path $CertPath.FullName -Filter '*.pfx'
+        if ($null -eq $pfxFiles) {
+            throw New-Object System.NullReferenceException("Unable to locate .pfx files under the specified CertPath location")
+        }
+
+        "Retrieving Rest Certificate" | Trace-Output
+        $currentRestCertificate = Get-SdnNetworkControllerRestCertificate
```

Comment:
`Get-SdnNetworkControllerRestCertificate` depends on Get-NetworkController output which will be failed in Cert expire case. Could this be updated to remove dependency on a working NC Cluster so that it can be re-used in cert expire case?

Reply:
Sure thing. Let me evaluate `Get-SdnNetworkControllerInfoOffline` and see what can be modified.

Reply:
`Get-SdnNetworkControllerRestCertificate` has built-in methods now that if NC is offline, it will automatically handle getting the appropriate data and returning back a consistent response similar to NC was operational.